src/docs/release/breaking-changes/hero-controller-scope.md

```diff
@@ -0,0 +1,141 @@
+
+## Description of change
+
+If the code starts throwing assertion error after this change,
+it means the code has already been broken even before this change.
+This change only surfaced the problem.
+
+
+## Migration guide
+
+An example application that starts to throw exceptions.
+
+<!-- skip -->
+```dart
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(
+    MaterialApp(
+      builder: (BuildContext context, Widget child) {
+        // Builds two parallel navigators. This will throw
+        // error because both of navigators are under the same
+        // hero controller scope created by MaterialApp.
+        return Stack(
+          children: <Widget>[
+            Navigator(
+              onGenerateRoute: (RouteSettings settings) {
+                return MaterialPageRoute<void>(
+                  settings: settings,
+                  builder: (BuildContext context) {
+                    return const Text('first Navigator');
+                  }
+                );
+              },
+            ),
+            Navigator(
+              onGenerateRoute: (RouteSettings settings) {
+                return MaterialPageRoute<void>(
+                  settings: settings,
+                  builder: (BuildContext context) {
+                    return const Text('Second Navigator');
+                  }
+                );
+              },
+            ),
+          ],
+        );
+      }
+    )
+  );
+}
+```
+
+You can fix this application by introducing your own hero controller scopes.
+
+<!-- skip -->
+```dart
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(
+    MaterialApp(
+      builder: (BuildContext context, Widget child) {
+        // Builds two parallel navigators.
+        return Stack(
+          children: <Widget>[
+            HeroControllerScope(
+              controller: HeroController(createRectTween: materialRectTween),
+              child: Navigator(
+                onGenerateRoute: (RouteSettings settings) {
+                  return MaterialPageRoute<void>(
+                    settings: settings,
+                    builder: (BuildContext context) {
+                      return const Text('first Navigator');
+                    }
+                  );
+                },
+              ),
+            ),
+            HeroControllerScope(
+              controller: HeroController(createRectTween: materialRectTween),
+              child: Navigator(
+                onGenerateRoute: (RouteSettings settings) {
+                  return MaterialPageRoute<void>(
+                    settings: settings,
+                    builder: (BuildContext context) {
+                      return const Text('second Navigator');
+                    }
+                  );
+                },
+              ),
+            ),
+          ],
+        );
+      }
+    )
+  );
+}
+
+RectTween materialRectTween(Rect begin, Rect end) {
+  return MaterialRectArcTween(begin: begin, end: end);
```

Comment:
we should add a static method to MaterialRectArcTween that does this so that we don't have to include it in the sample code here